service/src/main/java/com/kodemon/service/facade/PokemonFacadeImpl.java

```diff
@@ -37,4 +43,15 @@ public void renamePokemon(PokemonDTO pokemon, String newName) {
         LOG.debug("Renaming Pokemon {}. New name: {}", pokemon, newName);
         pokemonService.renamePokemon(beanMappingService.mapTo(pokemon, Pokemon.class), newName);
     }
+
+    @Override
+    public PokemonDTO generateWildPokemon(UserDTO user) {
+        Random rand = new Random();
```

Comment:
I think taht this is too much business logic for the facade layer. Cannot it be moved into Service somethow?